pkg/osquery/runtime/osquery_instance_history/history.go

```diff
@@ -0,0 +1,65 @@
+package osquery_instance_history
+
+import (
+	"errors"
+	"time"
+)
+
+const max_instances = 10
+
+type History struct {
+	instances []*Instance
+}
+
+// GetHistory returns the last 10 instances of osquery started / restarted by launcher
+// each start / restart cycle is an entry
+func GetHistory() []Instance {
+	results := []Instance{}
+	for _, v := range currentHistory.instances {
+		results = append(results, *v)
+	}
+	return results
+}
+
+// CurrentInstance returns the current osquery instance
+func CurrentInstance() Instance {
+	return *currentHistory.currentInstance()
+}
+
+func (h *History) nextInstance() error {
+	if h.currentInstance() != nil && h.currentInstance().ExitTime == "" {
+		return errors.New("cannot create new instance while current instance does not have exit time")
+	}
+
+	newInstance := &Instance{
+		StartTime: timeNow(),
+	}
+
+	h.setCurrentInstance(newInstance)
+
+	return nil
+}
+
+func (h *History) currentInstance() *Instance {
+	if h.instances != nil && len(h.instances) > 0 {
+		return h.instances[len(h.instances)-1]
+	}
+	return nil
+}
+
+func (h *History) setCurrentInstance(instance *Instance) {
```

Comment:
I might quibble over `add` vs `set`. But :shrug: quibbles

Reply:
yes, we should have a quibble session on naming before this goes out